Buildings/Controls/OBC/Facade/ShadingEnable.mo

```diff
@@ -0,0 +1,121 @@
+within Buildings.Controls.OBC.Facade;
+block ShadingEnable "Generic shading device enable/disable seqence"
+
+  parameter Boolean use_solIrr = false
+    "Set to true if enabling based on solar irradiance, otherwise enabling based on temeprature"
+    annotation(Dialog(group = "Conditional"));
+  parameter Modelica.SIunits.Temperature TThr = 298.15
+    "Temperature threshold (either zone or outdoor air)"
+    annotation(Evaluate=true, Dialog(group="Hysteresis", enable = not use_solIrr));
+  parameter Modelica.SIunits.Irradiance irrThr = 1000
+    "Solar irradiance threshold"
+    annotation(Evaluate=true, Dialog(group="Hysteresis", enable = use_solIrr));
+
+  CDL.Interfaces.BooleanInput uEnable "Shading device enable schedule"
+    annotation (Placement(transformation(extent={{-120,-60},{-80,-20}}),
+    iconTransformation(extent={{-140,-80},{-100,-40}})));
+  CDL.Interfaces.RealInput T if not use_solIrr
```

Comment:
It would be easier if there are two blocks, `Shade_T` and `Shade_H`, one for temperature and one for irradiation.